userland/libtock/tock.c

```diff
@@ -155,6 +155,19 @@ void* tock_app_grant_begins_at(void) {
   return memop(6, 0);
 }
 
+#pragma GCC diagnostic ignored "-Wbad-function-cast"
```

Comment:
You should wrap this is a `pragma push/pop` like we do elsewhere so that the rest of this file is still covered by this check. We really only want to disable it for this function